File: articles/collective-action-and-cooperation/fortran/collective_action_score.f90
program collective_action_score
  implicit none

  real :: incentive_alignment, trust, legitimacy, norm_strength
  real :: enforcement_credibility, communication_quality, coordination_quality
  real :: perceived_fairness, free_riding_pressure, burden_inequality
  real :: hypocrisy_visibility, scale_complexity, score

  incentive_alignment = 78.0
  trust = 76.0
  legitimacy = 74.0
  norm_strength = 72.0
  enforcement_credibility = 70.0
  communication_quality = 80.0
  coordination_quality = 75.0
  perceived_fairness = 73.0
  free_riding_pressure = 30.0
  burden_inequality = 25.0
  hypocrisy_visibility = 20.0
  scale_complexity = 35.0

  score = 0.12 * incentive_alignment + &
          0.13 * trust + &
          0.12 * legitimacy + &
          0.11 * norm_strength + &
          0.10 * enforcement_credibility + &
          0.11 * communication_quality + &
          0.11 * coordination_quality + &
          0.10 * perceived_fairness - &
          0.12 * free_riding_pressure - &
          0.07 * burden_inequality - &
          0.06 * hypocrisy_visibility - &
          0.05 * scale_complexity

  print *, "Collective action raw score:", score
end program collective_action_score

File: articles/collective-action-and-cooperation/go/collective_action_score.go
package main

import "fmt"

type CollectiveActionCase struct {
	IncentiveAlignment    float64
	Trust                 float64
	Legitimacy            float64
	NormStrength          float64
	EnforcementCredibility float64
	CommunicationQuality  float64
	CoordinationQuality   float64
	PerceivedFairness     float64
	FreeRidingPressure    float64
	BurdenInequality      float64
	HypocrisyVisibility   float64
	ScaleComplexity       float64
}

func CollectiveActionScoreRaw(x CollectiveActionCase) float64 {
	return 0.12*x.IncentiveAlignment +
		0.13*x.Trust +
		0.12*x.Legitimacy +
		0.11*x.NormStrength +
		0.10*x.EnforcementCredibility +
		0.11*x.CommunicationQuality +
		0.11*x.CoordinationQuality +
		0.10*x.PerceivedFairness -
		0.12*x.FreeRidingPressure -
		0.07*x.BurdenInequality -
		0.06*x.HypocrisyVisibility -
		0.05*x.ScaleComplexity
}

func main() {
	demo := CollectiveActionCase{78, 76, 74, 72, 70, 80, 75, 73, 30, 25, 20, 35}
	fmt.Printf("Collective action raw score: %.2f\n", CollectiveActionScoreRaw(demo))
}
